### ISSUE TITLE
batch normalization folding issue in Conv2d with dilation>1

### ISSUE THREAD

**Issue by elad-c** [tensorflow/tensorflow#50172]
**System information**
- OS Platform and Distribution: Linux Ubuntu 18.04
- TensorFlow version (use command below):
- Python version: v2.5.0-rc3-213-ga4dfb8d1a71 2.5.0

**Describe the current behavior**
I'm converting a simple Conv2d->BatchNorm->ReLU network to TFLite. In previous versions, the BatchNorm would be folded into the Conv2D weights to avoid the extra quantization. In TF 2.5.0, Conv2D with dilation>1 is applied with space2batch - batch2space nodes, which seems to confuse the converter.

**Describe the expected behavior**
The expected behaviour should be to fold the BatchNorm weights into the Conv2D weights (The same way as in TF 2.4.1)

**Standalone code to reproduce the issue**
```
import tensorflow as tf
import numpy as np


def representative_dataset():
    for _ in range(10):
        data = np.random.rand(1, 244, 244, 3)
        yield [data.astype(np.float32)]


dil = 2
for ptq in [False, True]:
    _in = tf.keras.Input((244, 244, 3))
    if ptq:
        x = _in
    else:
        x = tf.quantization.fake_quant_with_min_max_args(_in)

    x = tf.keras.layers.Conv2D(16, 3, padding='same', dilation_rate=(dil, dil), use_bias=False)(x)
    x = tf.keras.layers.BatchNormalization()(x)
    x = tf.keras.layers.ReLU()(x)
    if not ptq:
        x = tf.quantization.fake_quant_with_min_max_args(x)
    model = tf.keras.Model(_in, x)

    converter = tf.lite.TFLiteConverter.from_keras_model(model)
    converter.optimizations = [tf.lite.Optimize.DEFAULT]
    if ptq:
        converter.representative_dataset = representative_dataset
    tflite_quant_model = converter.convert()

    s = '_ptq' if ptq else ''
    tflite_file = f'/tmp/dilated_conv2d/dil{dil}{s}{c}.tflite'
    with open(tflite_file, 'wb') as f:
        f.write(tflite_quant_model)

    print(f'Generated {tflite_file} with TF {tf.__version__}')

```
TF 2.5.0: Quantizting with fake_quant nodes
![dil2](https://user-images.githubusercontent.com/78862769/121343752-b1e0ab80-c92b-11eb-88a7-5a53e5583b89.png)

TF 2.5.0: Quantizting without fake_quant nodes
![dil2_ptq](https://user-images.githubusercontent.com/78862769/121343757-b311d880-c92b-11eb-94e6-408be513cc00.png)

TF 2.4.1: Quantizting without fake_quant nodes
![dil2_ptq_TF241](https://user-images.githubusercontent.com/78862769/121343755-b2794200-c92b-11eb-83c9-b86e0e506d8b.png)

(Images generated with Netron: https://netron.app/)

**Comment by yyoon** [comment]
cc: @abattery @thaink @rino20 @ethkim
Could any of you triage this further? I'm not sure whether this is a general conversion issue or model optimization specific issue.

**Comment by abattery** [comment]
Looks like this is a newly introduced behavior by the MLIR quantizer in TF 2.5. @teijeong @liufengdb could you take a look at this?

**Comment by teijeong** [comment]
Thanks for spotting this. Meanwhile, you can specify `converter.experimental_new_quantizer = False` to fall back to old quantizer.

**Comment by teijeong** [comment]
Seems like regression in the converter. Even a float model generated from TF 2.5 has SpaceToBatchNd ops (and so). @renjie-liu, can you take a look?

![image](https://user-images.githubusercontent.com/4837376/122018099-82371500-cdfd-11eb-8ef2-0e444243ac99.png)

**Comment by haozha111** [comment]
Hi,

Sorry for late reply. I've verified this is causes by the dynamic batch size dimension in your model. Basically the dilated conv will fail to match at this line:
https://github.com/tensorflow/tensorflow/blob/7d94fc2b30f66cf2ee269b6cf42bdeef8f40a288/tensorflow/compiler/mlir/lite/transforms/dilated_conv.h#L109

The keras input layer will create an input with dynamic batch size, could you specify a fix batch size when building the input layer? Something like:
  inp = tf.keras.layers.Input(batch_size=1, shape=(224, 224, 3))

I've verified the dilated conv will be fused once changing to a fixed batch size.

**Comment by sachinprasadhs** [comment]
@elad-c , Could you please try specifying fixed batch size as per the above comment and try with the latest Tensorflow version. Thanks!
